### PR TITLE
testing: Update docker and docker-compose versions

### PR DESCRIPTION
**Pull request by marclop**
## What does this PR do?

Updates the `docker` and `docker-compose` to support newer versions of
the docker-compose file and newer docker-compose features like profiles.

## Why is it important?

To allow tests in projects that depend on beats, like APM Server to use more up-to-date versions of Docker Compose and leverage new docker-compose features.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

N/A.

## How to test this PR locally

Ensure that `pip -r libbeat/tests/system/requirements.txt` works in a venv.

## Related issues

Unblocks elastic/apm-server#7453

## Use cases

Use [`docker-compose --profile`](https://docs.docker.com/compose/profiles/) to enable docker-compose files to have disabled services by default, yet allow them to be spun up by specifying a profile.

### CODE REVIEW

**Comment by marclop** [comment]
I had to modify `metricbeat/module/prometheus/docker-compose.yml` to use a `prometheus_base` service which is extended by two services used in the Metricbeat tests. The reason is that newer versions of Docker Compose are more strict in what's set in the `docker-compose.yml`.

The problem here was that the `prometheus` service explicitly exported the `9090/tcp` port and then `prometheus-host-network` service, extended it and set the network to `host` which already published all ports since the network is the same as the host.

The actual error was: `docker.errors.InvalidArgument: "host" network_mode is incompatible with port_bindings`.

**Review by aspacca — APPROVED** [review]

**Review by simitt — APPROVED** [review]
LGTM - but please wait for a review from the elastic-agent-control-plane team

**Review by belimawr — APPROVED** [review]
LGTM

**Comment by ph** [comment]
LGTM @marclop

**Pull request merged**
